src/treeffuser/_score_models.py

```diff
@@ -107,8 +107,9 @@ def _make_training_data(
     train_mean, train_std = sde.get_mean_std_pt_given_y0(y_train, t_train)
     perturbed_y_train = train_mean + train_std * z_train
 
-    predictors_train = np.concatenate([X_train, perturbed_y_train, t_train], axis=1)
+    predictors_train = np.concatenate([perturbed_y_train, X_train, t_train], axis=1)
     predicted_train = -1.0 * z_train
+    cat_idx = [c + y_train.shape[1] for c in cat_idx] if cat_idx is not None else None
```

Comment:
yes, this should be here and not in the main script. thanks for adding y_dim too